src/Async.js

```diff
@@ -85,6 +86,9 @@ const Async = React.createClass({
 				cache: initCache(nextProps.cache),
 			});
 		}
+		if (nextProps.value !== this.props.value) {
+			this.loadOptions('');
```

Comment:
Ah, yes you are quite right. But searching for anything else in the `this.loadOptions` call won't change this, since as you rightly noticed above - `value` is the current value of the select box, not the search query string!

The way I get around this in my implementation is this:

===================================================
Presumably, where I render a react-select component, to be able to pass the `value` prop I must also have access to `value` when I pass the `loadOptions` prop. So I can simply always add that 101st record to the `loadOptions` return value, by doing a simple findWhere. E.g.

```js
export default class SomeComponent extends React.Component {
  static propTypes = {
    items: React.PropTypes.array,
    someValue: React.PropTypes.number
  };

  render () {
    return (
      <div>
        /* ...other stuff */
        <Select.Async
          value={this.props.someValue}
          loadOptions={::this.getItems}
        />
      </div>
    )
  }

  getItems (query, callback) {
    const itemsFromQuery = this.itemsFromQuery(query)
    callback(null, { options: itemsFromQuery })
  }

  itemsFromQuery (query) {
    const { items, someValue } = this.props
    const currentItem = _.findWhere(items, { value: someValue })
    const filteredItems = items.filter(() => {
      // ...do filtering logic based on passed 'query' string
    })
    filteredItems.unshift(currentItem) // add 101st item 
    return filteredItems
  }
}
```

Since the limiting / filtering logic happens outside of the creation of the `Select.Async` component (in our own wrapper, like in my above example) `Select.Async` shouldn't handle the re-adding of the current item, our external wrapper should handle it. Limiting functionality should be added to react-select core code itself otherwise.

Normally if the `Select.Async` `componentWillReceiveProps` method is being called, then it means that an item has been selected, meaning we want to refresh the list. If there are scenarios outside of this you can think of, meaning we need to instead pass the original query value instead of '' then let me know!